### PR TITLE
feat: Autoreload

### PR DESCRIPTION
**Pull request by theduke**
- docs: Add hello-world example
- feat: Use clap for CLI argument parsing
- feat: Javascript code autoreload
- build: Nix devshell improvements

### CODE REVIEW

File: src/ion_runner/mod.rs [hunk 1]
```diff
@@ -313,6 +316,8 @@ pub struct IonRunner {
     user_code: String,
 }
 
+pub type SharedIronRunner = Arc<Mutex<IonRunner>>;
```
Comment (Arshia001) [inline src/ion_runner/mod.rs:319]: It's 'Ion', not Iron.

File: src/main.rs [hunk 2]
```diff
@@ -39,35 +44,113 @@ async fn main() {
...
+/// Start a WinterJS webserver serving the given JS app.
+#[derive(clap::Parser, Debug)]
+struct CmdServe {
+    /// The port to listen on.
+    #[clap(short, long, env = "WINTERJS_PORT")]
+    port: Option<u16>,
+
+    /// The interface to listen on.
+    /// Defaults to 127.0.0.1
+    #[clap(long, default_value = "127.0.0.1", env = "WINTERJS_INTERFACE")]
```
Comment (Arshia001) [inline src/main.rs:142]: I believe 'IP' or 'listen IP' would be the more obvious name.

File: src/main.rs [hunk 3]
```diff
@@ -39,35 +44,113 @@ async fn main() {
...
+                format!(
+                    "Could not read Javascript file at '{}'",
+                    cmd.js_path.display()
+                )
+            })?;
 
-        std::fs::read_to_string(&path)
-            .with_context(|| format!("Could not read js file at '{}'", path))?
-    };
+            // let listen = cmd.liste
```
Comment (Arshia001) [inline src/main.rs:79]: Nit

**Review by Arshia001 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (theduke) [reply src/main.rs]: Changed to _IP

File: src/ion_runner/mod.rs (hunk 1 above)
Comment (theduke) [reply src/ion_runner/mod.rs]: Fixed.

**Review by Arshia001 — APPROVED** [review]

**Pull request merged**